lib/icinga/downtime.cpp

```diff
@@ -427,6 +443,26 @@ bool Downtime::CanBeTriggered()
 	return true;
 }
 
+void Downtime::SetupCleanupTimer()
+{
+	if (!m_CleanupTimer) {
+		m_CleanupTimer = Timer::Create();
+
+		auto name (GetName());
+
+		m_CleanupTimer->OnTimerExpired.connect([name=std::move(name)](const Timer * const&) {
+			auto downtime (Downtime::GetByName(name));
```

Comment:
Is the downtime captured by name here to simulate the behavior of a `weak_ptr`?

Reply:
Yes, such don’t exist for intrusive ones IIRC.